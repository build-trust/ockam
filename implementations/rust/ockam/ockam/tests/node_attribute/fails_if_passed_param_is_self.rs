// This test checks that #[ockam::node] causes a compile time error
// if the function is passed a `self` param (thus making it a
// `Receiver` function.

#[ockam::node]
async fn main(self) {}
