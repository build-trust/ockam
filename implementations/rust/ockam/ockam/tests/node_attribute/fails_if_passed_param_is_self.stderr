error: Input argument should be of type `ockam::Context`
 --> $DIR/fails_if_passed_param_is_self.rs:6:15
  |
6 | async fn main(self) {}
  |               ^^^^

error[E0601]: `main` function not found in crate `$CRATE`
 --> $DIR/fails_if_passed_param_is_self.rs:5:1
  |
5 | / #[ockam::node]
6 | | async fn main(self) {}
  | |______________________^ consider adding a `main` function to `$DIR/tests/node_attribute/fails_if_passed_param_is_self.rs`
